bin/reth/src/args/network_args.rs

```diff
@@ -52,11 +52,12 @@ impl NetworkArgs {
     /// values in this option struct.
     pub fn network_config(&self, config: &Config, chain_spec: ChainSpec) -> NetworkConfigBuilder {
         let peers_file = (!self.no_persist_peers).then_some(&self.peers_file);
-        config
+        let network_config_builder = config
             .network_config(self.nat, peers_file.map(|f| f.as_ref().to_path_buf()))
             .boot_nodes(self.bootnodes.clone().unwrap_or_else(mainnet_nodes))
-            .chain_spec(chain_spec)
-            .set_discovery(self.disable_discovery)
+            .chain_spec(chain_spec);
+
+        self.discovery.apply_to_builder(network_config_builder)
```

Comment:
not a fan of this pattern in general.

instead, the `network_config_builder` should consume all fields of the DiscoveryArgs.
There should already be functions for configuring this on the networkconfig builder

Reply:
Meh. I agree but it seems fine at the moment, given the helper already does what you said.